test/suites/secondary_http.bash

```diff
@@ -0,0 +1,89 @@
+start_http_server() {
+    local HOST="127.0.0.1"
+    local PORT="8080"
+    local URL="http://${HOST}:${PORT}/"
+    export CCACHE_SECONDARY_STORAGE="${URL}"
+
+    trap 'terminate_all_children' EXIT # also cleanup after exceptional code flow
```

Comment:
Good idea to clean up after e.g. ctrl+c! I suggest moving the `trap` statement to `test/run` since there can be only one EXIT handler.

Maybe this also removes the need of a teardown mechanism? `test/run` could call `terminate_all_children` after each suite.

Reply:
The `TEST` / `reset_environment` removes the cwd from the HTTP server. Therefore I need to restart the server.